src/main/scala/org/constellation/domain/redownload/RedownloadService.scala

```diff
@@ -70,19 +70,33 @@ class RedownloadService[F[_]](
       }
     } yield proposals
 
+  def fetchMissingSnapshots: F[SnapshotsAtHeight] = ??? // TODO: Implement
+
   def checkForAlignmentWithMajoritySnapshot(): F[Unit] =
     for {
       peersProposals <- getPeerProposals()
       createdSnapshots <- getCreatedSnapshots()
+      acceptedSnapshots <- getAcceptedSnapshots()
+
       majorityState = majorityStateChooser.chooseMajorityState(
         createdSnapshots,
         peersProposals
       )
-      // TODO: Calculate diff and call redownload if true
-      _ <- if (shouldRedownload(createdSnapshots, majorityState, snapshotHeightRedownloadDelayInterval)) F.unit
-      else F.unit
+
+      _ <- if (shouldRedownload(acceptedSnapshots, majorityState, snapshotHeightRedownloadDelayInterval)) {
+        val plan = calculateRedownloadPlan(acceptedSnapshots, majorityState)
+        applyRedownloadPlan(plan) // TODO: Actual redownload, rethink splitting the logic
+      } else F.unit
     } yield ()
 
+  private[redownload] def applyRedownloadPlan(plan: RedownloadPlan): F[Unit] = for {
+    missingSnapshots <- fetchMissingSnapshots // TODO: Store these snapshots on disk now?
+    _ <- acceptedSnapshots.modify { m =>
+      val updated = plan.expectedResult // TODO: Is it the correct way?
```

Comment:
Yes, I think so :)
Anyway, we can keep it like this and once we'll implement the full redownload logic, let's back to it and maybe we'll be able to check what's currently accepted (or maybe re-verify this `expectedResult` as a sanity check, that all expected data are stored properly)